plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/CostModelInterface.hs

```diff
@@ -101,10 +93,10 @@ applyParams cm params = case toJSON cm of
     Object o ->
         let
             hashmapified = HM.fromList $ Map.toList params
-            scaledNumbers = fmap (\n -> Number $ fromIntegral n / scaleFactor) hashmapified
+            usingScientific = fmap (Number . fromIntegral) hashmapified
             flattened = flattenObject "-" o
             -- this is where the overwriting happens, this is left-biased
-            merged = HM.union scaledNumbers flattened
+            merged = HM.union usingScientific flattened
             unflattened = unflattenObject "-" merged
         in case fromJSON (Object unflattened) of
```

Comment:
I was tempted to make this point-free for maximum incomprehensibility, but thought better of it.